dot/network/transaction.go

```diff
@@ -107,58 +110,60 @@ func decodeTransactionHandshake(_ []byte) (Handshake, error) {
 	return &transactionHandshake{}, nil
 }
 
-func (s *Service) createBatchMessageHandler(txnBatchCh chan *BatchMessage) NotificationsMessageBatchHandler {
-	go func() {
-		protocolID := s.host.protocolID + transactionsID
-		ticker := time.NewTicker(s.cfg.SlotDuration)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case <-s.ctx.Done():
-				return
-			case <-ticker.C:
-				timeOut := time.NewTimer(s.cfg.SlotDuration / 3)
-				var completed bool
-				for !completed {
-					select {
-					case <-timeOut.C:
-						completed = true
-					case txnMsg := <-txnBatchCh:
-						propagate, err := s.handleTransactionMessage(txnMsg.peer, txnMsg.msg)
-						if err != nil {
-							s.host.closeProtocolStream(protocolID, txnMsg.peer)
-							continue
-						}
-
-						if s.noGossip || !propagate {
-							continue
-						}
-
-						if !s.gossip.hasSeen(txnMsg.msg) {
-							s.broadcastExcluding(s.notificationsProtocols[TransactionMsgType], txnMsg.peer, txnMsg.msg)
-						}
+func (s *Service) startTxnBatchProcessing(txnBatchCh chan *BatchMessage) {
+	protocolID := s.host.protocolID + transactionsID
+	ticker := time.NewTicker(s.cfg.SlotDuration)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-s.ctx.Done():
+			return
+		case <-ticker.C:
+			timeOut := time.NewTimer(s.cfg.SlotDuration / 3)
+			var completed bool
+			for !completed {
```

Comment:
nit you could rename `completed` to `timedOut`

Reply:
Done